src/test/java/com/microsoft/sqlserver/jdbc/resultset/ResultSetTest.java

```diff
@@ -329,4 +329,141 @@ public void testGetterOnNull() throws SQLException {
             assertEquals(null, rs.getTime(1));
         }
     }
+
+    @Test
+    public void testHoldability() throws SQLException {
+        int[] holdabilityOptions = {ResultSet.HOLD_CURSORS_OVER_COMMIT, ResultSet.CLOSE_CURSORS_AT_COMMIT};
+
+        try (Connection con = DriverManager.getConnection(connectionString); Statement stmt = con.createStatement();
+                ResultSet rs = stmt.executeQuery("select null")) {
+
+            int connHold = con.getHoldability();
+            assertEquals(stmt.getResultSetHoldability(), connHold);
+            assertEquals(rs.getHoldability(), connHold);
+
+            for (int i = 0; i < holdabilityOptions.length; i++) {
+
+                if ((connHold = con.getHoldability()) != holdabilityOptions[i]) {
+                    con.setHoldability(holdabilityOptions[i]);
+                    assertEquals(con.getHoldability(), holdabilityOptions[i]);
```

Comment:
The purpose of setting holdability on a connection is so that it persists when retrieving result sets from said connection. Verifying setting/getting holdability on a connection object is a test with a pretty low chance of failure. Can we also create a ResultSet from the Connection here and verify the connection's holdability did in fact persist? Even better if we actually add scenarios where holdability behaves as expected instead of just checking getters. The latter can probably be done as an improvement though.

Reply:
that is tested elsewhere this test just tests getter/setter